### ISSUE TITLE
[Cauldron server with plugins] Asteroids one point of spawn, missing rockets

### ISSUE THREAD

**Issue by vadimyer** [micdoodle8/Galacticraft#2054]
My players fly on Asteroids very often, and almost 100% of them are missing a module with a rocket when they come there. Also, all of them, no matter what point of departure on the Earth was, arrive at one point on Asteroids. It seems like a platform built by some other player, it was once privatized as a WG region, I even removed that region and let it be for a few days, but still players are missing their rockets. I don't have any predefined spawn points there, no plugin seems to control that as well.

How do I know the reason?

**Comment by radfast** [comment]
Update Galacticraft.

**Comment by vadimyer** [comment]
@radfast I've updated GC to the latest available version from Jenkins, deleted some GC files from world/data folder. After that I could eventually land on some random asteroids with a module, but players are still having the same issue. Maybe I should wipe Asteroids dimensions totally?

**Comment by radfast** [comment]
Current GC:
* every visit to the asteroids should land on a different asteroid (starting with the closest - so all asteroids will be nearby)
* the game tracks which asteroids were already landed on in ```.minecraft/world/data/GCAsteroidSaveData.dat``` - if you delete that it should automatically generate a "clean" file
* every time there must be an Entry Pod containing the rocket item

You need to be more specific about what issue your players are having. 
- Are you saying they have an Entry Pod, but it contains no items?
- Or there is no Entry Pod?
- Is it the same issue as #1195?
- Are players having time-outs or disconnects while traveling to asteroids?  Flying up in the rocket - in the Celestial Catalog screen - or on the descent?

**Comment by vadimyer** [comment]
Hey,

This is what happens, and I think it's not #1195, because no one has timeouts or disconnects:

1) A player is going to the Asteroids, 3 tier rocket, starting from Earth. Then they select Asteroids in GC universe screen.
2) When they reach the dimension, they land without a module at one exact point, it's a platform built by some other player, but already without any WG defined region. So no random points of landing. Just that platform. There's no entry pod around (if I understand correctly, that's a "module").
3) As I've stated, I've deleted all GC***.dat files from world/data.
4) And eventually that helped for the first time (I've tested it only on myself). But after a server restart problem appeared again...

**Comment by radfast** [comment]
OK I think I understand the problem.  2 seems like a big issue.   I don't understand why nobody reported it before.

I can test multiplayer tonight ...

**Comment by radfast** [comment]
Please provide a screenshot of the platform they land on.

There is one situation where the game will "create" a small platform if it did not find any asteroid.  This could indicate a bug.

**Comment by vadimyer** [comment]
https://s.vadimyer.com/2016-02-04_15.26.55.png
https://s.vadimyer.com/2016-02-04_15.27.13.png

**Comment by vadimyer** [comment]
I've removed a platform and now I always land in an empty space, exactly where it was ._.

**Comment by vadimyer** [comment]
It seems like a landing point is always -256 Y -256, if it can say something.

**Comment by radfast** [comment]
It's not normal.

Have you installed any Add-Ons (Galaxy Space, More Planets, 4Space, Extra Planets ...)?

**Comment by vadimyer** [comment]
No, this server never had any addons installed.

**Comment by vadimyer** [comment]
._. That was because of a WorldBorder plugin. The Overworld has 10000x10000 size, but the Asteroids dimensions borders were set to 3000x3000, I remember that limiting planets with this plugin lesser than Overworld's size causes bugs, GC tries to land a player in a zone out of borders, so the module with a rocket is out.

But I've set the Asteroids world size to 10000x10000, same as Overworld, the issue persists. Okay, I've deleted the borders from this world totally. And now this:
https://s.vadimyer.com/e51090ea-4848-4fe0-8724-b2b85f8b3044.png

Please look at the coordinates. On Moon&Mars everything's OK with borders, players land within a radius of 10000x10000, but what about this on Asteroids?

**Comment by radfast** [comment]
It's not possible.

If players ride rockets, Galacticraft tries to keep the players at similar co-ordinates:
eg. 
Overworld (2000,64,1000) -> Moon (2000,64,1000)
Overworld (2000,64,1000) -> Mars (2000,80,1000)
Overworld (2000,64,1000) -> Asteroids (2036,134,990)

In Asteroids it can be not "exactly" the same (x, y, z) coordinates because it searches for the closest asteroid - it should always land on the centre of an asteroid.

https://github.com/micdoodle8/Galacticraft/blob/master/src/main/java/micdoodle8/mods/galacticraft/planets/asteroids/dimension/TeleportTypeAsteroids.java#L41

You can switch on debug in ```.minecraft/config/Galacticraft/core.conf```  set:
>     B:"Enable Debug Messages"=true

This will give more info in your server log which can maybe help with this problem.

**Comment by vadimyer** [comment]
I think that's way too much, I mean, the launch point in Overworld was like -1010, 70, 1500, but the landing point in Asteroids is 400000, 225, 2000000? :D

**Comment by vadimyer** [comment]
I turned on debug info and this is what I got:
```
 Found matching world (-30) for name: planet.asteroids
[18:57:58 INFO]: DEBUG: Attempting to remove player from old dimension 0
[18:57:58 INFO]: DEBUG: Successfully removed player from old dimension 0
[18:57:58 INFO]: DEBUG: Sending respawn packet to player for dim -30
[18:57:58 INFO]: Testing asteroid at x593576 y113 z574080
[18:57:58 INFO]: Found asteroid at x593576 z574080
[18:57:58 INFO]: DEBUG: Loading first chunk in new dimension.
```
Start points were +1795, 70, +274. Please consider tuning this, as on a busy server generating asteroids causes lots of lag ._.

**Comment by radfast** [comment]
I've looked at this.  It must be happening because you deleted data files.  I guess the server cannot regenerate all the data needed for asteroid landing, if the chunks are already fully generated.  I don't know why it's so crazy huge at 593576, 574080 but maybe somehow that's the only data remaining.

Suggestions:

1.  Delete all / most of the region files in the asteroids world folder.  Also delete the gcasteroidsavedata.dat file.
2.  Now go to the asteroids dimension (use /dimensiontp command), you should arrive close to 0,0.  Fly around in creative for a few minutes, to regenerate some map chunks.
3.  The server should fill a new gcasteroidsavedata.dat with sensible numbers matching the chunks you just generated.

If you have players with important structures / bases, you could leave some of the asteroids chunk files intact in the world folder, just make sure to fly around nearby to regenerate plenty of map.

**Comment by vadimyer** [comment]
@radfast here's what I did:

1) I've completely deleted DIM-30 and GCAsteroids .dat file.
2) Did that. Arrived to +2332, 144, +2312, points that were very close to my overworld launching point.
3) I was flying there for a few minutes, generating chunks. I think I've generated a pretty huge area around the point of my first arrival.
4) Went back to /spawn, teleported back to asteroid with /dimensiontp **from the same point** and:

```
Testing asteroid at x37354 y94 z36944
Found asteroid at x37354 z36944
```

You know what happened later. OK, I've started from scratch again, even tried to /dimensiontp to asteroids without coming back to Earth, but those 37354 95 36944 seem to happen every single time, after the first arrival.

**Comment by radfast** [comment]
Thanks for the updated info.  I guess I need to re-open this.  I still can see no reason for this.  Seems like a lot of problems on your server!

**Comment by DraksterAU** [comment]
I believe this is caused by using a Bukkit spawn handler (essentials or multiworld/multiverse?), it causes all players to enter the asteroids dimension at  the same place.

Solution for us was to create a spawn area.... That players can open containers in but can't place anything.

Hope this helps?

**Comment by DraksterAU** [comment]
PS if you mess with the priorities in the spawn handler plugin config you might be able to allow gc to override it..... Or gc can change the priority for the spawn / clone event when players enter the asteroids dimension to override all.
 
If im way off... Apologies

**Comment by SkyFire55** [comment]
I have exactly the same problem - spawning coordinates are huge. First 5-7 flights I don't have any problems, but later landing coords are not connected with starting coords.  Problem is not connected with plugins - I tested server with only 3 mods, which are needed for GC, and without any plugins. Server core - Thermos 1.7.10 1614, and any GC version show the same bug. Problem connected only with Asteroids.

**Comment by radfast** [comment]
Hi @SkyFire55 

If you're still playing Galacticraft :) ... there's a good chance this is fixed by a6c19fa in build 489+ of Galacticraft 3 (and also in build 72+ of Galacticraft 4).

Please try it out in a new world or at least a new DIM-30.  If it doesn't seem fixed to you,  please reopen this.  If still seeing the problem, please upload your ```GCAsteroidData.dat``` for me to look at - it's in the ```data``` folder in your world's main save folder.  Main save folder = the folder with all the different dimensions like DIM1 and DIM-30.
